services/src/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/SwitchActivationSpec.groovy

```diff
@@ -0,0 +1,158 @@
+            !Cookie.isDefaultRule(it.cookie)
+        }*.cookie
+        assert createdCookies.size() == 2
+
+        northbound.deleteSwitchRules(switchPair.src.dpId, DeleteRulesAction.IGNORE_DEFAULTS)
+        Wrappers.wait(WAIT_OFFSET) {
+            verifyAll(northbound.validateSwitch(switchPair.src.dpId)) {
+                it.rules.missing.containsAll(createdCookies)
+                switchHelper.verifyRuleSectionsAreEmpty(it, ["proper", "excess"])
+            }
+        }
+
+        lockKeeper.knockoutSwitch(switchPair.src)
+        Wrappers.wait(WAIT_OFFSET) { assert !(switchPair.src.dpId in northbound.getActiveSwitches()*.switchId) }
+
+        when: "Connect the switch to the controller"
+        lockKeeper.reviveSwitch(switchPair.src)
+        Wrappers.wait(WAIT_OFFSET) { assert switchPair.src.dpId in northbound.getActiveSwitches()*.switchId }
+
+        then: "Missing flow rules were synced during switch activation"
+        verifyAll(northbound.validateSwitch(switchPair.src.dpId)) {
+            it.rules.proper.containsAll(createdCookies)
+            switchHelper.verifyRuleSectionsAreEmpty(it, ["missing", "excess"])
+        }
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "Missing flow meters are synced from a new switch connected to the controller"() {
+        given: "A switch with missing flow meters and not connected to the controller"
+        def sw = topology.getActiveSwitches().find { it.virtual }
+        def flow = flowHelper.singleSwitchFlow(sw)
+        flowHelper.addFlow(flow)
+        def createdMeterIds = northbound.getAllMeters(sw.dpId).meterEntries.findAll {
+            it.meterId > MAX_SYSTEM_RULE_METER_ID
+        }*.meterId
+        createdMeterIds.each { northbound.deleteMeter(sw.dpId, it) }
+
+        Wrappers.wait(WAIT_OFFSET) {
+            verifyAll(northbound.validateSwitch(sw.dpId)) {
+                it.meters.missing.size() == 2
+                switchHelper.verifyMeterSectionsAreEmpty(it, ["excess", "proper", "misconfigured"])
+            }
+        }
+
+        lockKeeper.knockoutSwitch(sw)
+        Wrappers.wait(WAIT_OFFSET) { assert !(sw.dpId in northbound.getActiveSwitches()*.switchId) }
+
+        when: "Connect the switch to the controller"
+        lockKeeper.reviveSwitch(sw)
+        Wrappers.wait(WAIT_OFFSET) { assert sw.dpId in northbound.getActiveSwitches()*.switchId }
+
+        then: "Missing flow meters were synced during switch activation"
+        verifyAll(northbound.validateSwitch(sw.dpId)) {
+            it.meters.proper*.meterId.sort() == createdMeterIds.sort()
+            switchHelper.verifyMeterSectionsAreEmpty(it, ["excess", "missing", "misconfigured"])
+        }
+
+        and: "Cleanup: Delete the flow"
+        flowHelper.deleteFlow(flow.id)
+    }
+
+    def "Excess rules/meters are synced from a new switch connected to the controller"() {
```

Comment:
I'd say '...upon connecting to the controller' or '...before connecting..' or '..when connecting..'

Reply:
changed to `Excess rules/meters are synced from a new switch before connecting`

Reply:
I don't see the change

Reply:
I don't have idea how it happened
sorry for any inconvenience
now it is totally fixed